arbos/block_processor.go

```diff
@@ -115,65 +115,65 @@ func createNewHeader(prevHeader *types.Header, l1info *L1Info, state *arbosState
 
 type ConditionalOptionsForTx []*arbitrum_types.ConditionalOptions
 
-type SequencingHooks struct {
-	NextTxToSequence        func() (*types.Transaction, error)                                                                                                                                      // Must be set
-	SequencedTx             func(int) (*types.Transaction, error)                                                                                                                                   // Must be set
-	TxErrors                []error                                                                                                                                                                 // This can be unset
-	DiscardInvalidTxsEarly  bool                                                                                                                                                                    // This can be unset
-	PreTxFilter             func(*params.ChainConfig, *types.Header, *state.StateDB, *arbosState.ArbosState, *types.Transaction, *arbitrum_types.ConditionalOptions, common.Address, *L1Info) error // This has to be set. Writes to *state.StateDB object should be avoided to prevent invalid state from permeating
-	PostTxFilter            func(*types.Header, *state.StateDB, *arbosState.ArbosState, *types.Transaction, common.Address, uint64, *core.ExecutionResult) error                                    // This has to be set
-	BlockFilter             func(*types.Header, *state.StateDB, types.Transactions, types.Receipts) error                                                                                           // This can be unset
-	ConditionalOptionsForTx []*arbitrum_types.ConditionalOptions                                                                                                                                    // This can be unset
+type SequencingHooks interface {
+	NextTxToSequence() (*types.Transaction, *arbitrum_types.ConditionalOptions, error)
+	DiscardInvalidTxsEarly() bool
+	SequencedTx(int) (*types.Transaction, error)
```

Comment:
I think SequencedTx now can be removed from the interface? (we can keep it if needed.. but generally speaking smaller interface is better)

Reply:
Done.